commit 595b893e2087de306d0781795fb8ec47873596a6
Author: Kees Cook <kees@kernel.org>
Date:   Tue May 3 13:55:00 2022 -0700

    randstruct: Reorganize Kconfigs and attribute macros
    
    In preparation for Clang supporting randstruct, reorganize the Kconfigs,
    move the attribute macros, and generalize the feature to be named
    CONFIG_RANDSTRUCT for on/off, CONFIG_RANDSTRUCT_FULL for the full
    randomization mode, and CONFIG_RANDSTRUCT_PERFORMANCE for the cache-line
    sized mode.
    
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20220503205503.3054173-4-keescook@chromium.org

diff --git a/include/linux/compiler-gcc.h b/include/linux/compiler-gcc.h
index 52299c957c98..a0c55eeaeaf1 100644
--- a/include/linux/compiler-gcc.h
+++ b/include/linux/compiler-gcc.h
@@ -66,14 +66,6 @@
 		__builtin_unreachable();	\
 	} while (0)
 
-#if defined(RANDSTRUCT_PLUGIN) && !defined(__CHECKER__)
-#define __randomize_layout __attribute__((randomize_layout))
-#define __no_randomize_layout __attribute__((no_randomize_layout))
-/* This anon struct can add padding, so only enable it under randstruct. */
-#define randomized_struct_fields_start	struct {
-#define randomized_struct_fields_end	} __randomize_layout;
-#endif
-
 /*
  * GCC 'asm goto' miscompiles certain code sequences:
  *